### PR TITLE
Fix missing Content-Range on replica query-link range responses

### LINKED ISSUE #1328: Replica query link returns 206 without Content-Range (breaks Foxglove MCAP seek/backfill)
## Bug report

Opening MCAP files from a generated query link in Foxglove fails with:

`Error at state seek-backfill: Record content length 12747022565183126887 is too large`

The same file works if downloaded first and then opened locally in Foxglove.

## Reproduction

1. Generate a query link to an MCAP file via replica API (example from `play.reduct.store/replica/api/v1/links/...`).
2. Open that URL directly in Foxglove.
3. Foxglove fails during seek/backfill.

## Observed HTTP behavior (from browser HAR)

Range requests are sent by client:
- `Range: bytes=0-120366711`
- `Range: bytes=120366675-120366711`
- etc.

Server responses:
- Status `206 Partial Content`
- `Content-Type: application/mcap`
- `Accept-Ranges: bytes`
- **Missing `Content-Range` header on all 206 responses**

This appears to violate partial content expectations and breaks Foxglove's random-access parsing for MCAP.

## Expected behavior

For every `206` response, include:
- `Content-Range: bytes <start>-<end>/<total>`
- `Content-Length: <end-start+1>`
- `Accept-Ranges: bytes`

## Notes

- Downloading the same URL to a local file and opening it works, so MCAP content itself appears valid.
- Problem is specific to HTTP range semantics on replica query link endpoint.

### PR DESCRIPTION
**Pull request by mother-6000**
Closes #1328

### Please check if the PR fulfills these requirements

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] CHANGELOG.md has been updated (for bug fixes / features / docs)

### What kind of change does this PR introduce?

Bug fix (HTTP range response compliance for replica query links).

### What was changed?

- Updated `GET /api/v1/links/:file_name` query-link range response handling to include `Content-Range` on `206 Partial Content` responses.
- Removed incorrectly echoed `Range` response header for ranged requests.
- Added helper logic to resolve range bounds for `Content-Range` formatting.
- Extended range test coverage to assert `content-range: bytes 0-2/6`.

### Related issues

- https://github.com/reductstore/reductstore/issues/1328
- Plan comment: https://github.com/reductstore/reductstore/issues/1328#issuecomment-4241736726

### Does this PR introduce a breaking change?

No.

### Other information:

Validation run locally:

- `cargo fmt --all`
- `cargo check --workspace`
- `cargo test -p reductstore test_get_query_link_range -- --nocapture`

### CODE REVIEW

**Comment by mother-6000** [comment]
@reductstore/maintainers Plan-approved implementation is ready for review. This PR addresses #1328 and follows the approved plan comment.

File: reductstore/src/api/http/links/get.rs [hunk 1]
```diff
@@ -242,6 +242,14 @@ async fn prepare_response(
             .satisfiable_ranges(initial_content_length)
             .collect::<VecDeque<_>>();
 
+        if ranges.is_empty() {
+            headers.insert(
```
Comment (atimin) [inline reductstore/src/api/http/links/get.rs:246]: test this branch

File: reductstore/src/api/http/links/get.rs [hunk 2]
```diff
@@ -274,6 +294,18 @@ async fn prepare_response(
     }
 }
 
+fn resolve_content_range(
```
Comment (atimin) [inline reductstore/src/api/http/links/get.rs:302]: cover with tests.

**Review by atimin — CHANGES_REQUESTED** [review]
Looks good, address my comments and update the change log.

**Review by atimin — APPROVED** [review]

File: reductstore/src/api/http/links/get.rs (hunk 1 above)
Comment (mother-6000) [reply reductstore/src/api/http/links/get.rs:246]: Added explicit coverage for this not-satisfiable branch in `test_get_query_link_range_not_satisfiable` (expects `416` and `content-range: bytes */6`). Re-ran the range tests locally and they pass.

File: reductstore/src/api/http/links/get.rs (hunk 2 above)
Comment (mother-6000) [reply reductstore/src/api/http/links/get.rs:302]: Added dedicated unit coverage for `resolve_content_range` in `test_resolve_content_range_variants`, including bounded, open-ended, and invalid/excluded cases. The test passes locally.

**Comment by mother-6000** [comment]
Addressed requested updates:
- Added CHANGELOG entry (commit 856255a).
- Added/extended tests for range edge cases and `resolve_content_range` coverage (commits eb8ddbc, d5d9fd4).

Local validation re-run:
- `cargo fmt --all`
- `cargo check --workspace`
- `cargo test -p reductstore test_get_query_link_range -- --nocapture`
- `cargo test -p reductstore test_get_query_link_range_not_satisfiable -- --nocapture`
- `cargo test -p reductstore test_resolve_content_range_variants -- --nocapture`

CI run `24388390762` is green. Re-requesting review now.
